### PR TITLE
UserTag intermediate model

### PR DESCRIPTION
**Pull request by Magnico**
Changes made:
- Created model UserTag
- Created endpoints for tagging a user, retrieving the tagged users and posts.
- Tests for endpoints created

### CODE REVIEW

File: blog/models.py [hunk 1]
```diff
@@ -9,6 +9,15 @@ class Post(models.Model):
...
     safe = models.BooleanField(default=True)
+    tagged_users = models.ManyToManyField(User, through='UserTag', related_name='tagged_users')
+
+    @property
+    def tagged_count(self):
+        return self.tagged_users.count()
+    
+    @property
+    def last_tag_date(self):
+        return self.usertag_set.last().created_at if self.usertag_set.last() else None
```
Comment (ejariza-evowill) [inline blog/models.py:20]: By truthy value, I think that you can use the `self.usertag_set` without the `last()` as well, but this is ok.``

File: blog/tests.py [hunk 2]
```diff
@@ -142,6 +141,47 @@ def test_post_comments_count(self):
...
+        #Check initial value of last_tag_date
+        Post.objects.create(title="test1", body="test1", author=self.user)
+        post = Post.objects.get(title="test1")
+        post.save()
+        self.assertEqual(post.last_tag_date, None)
+
+        #Creating and tagging 6 users
+        for i in range(6):
+            user = User.objects.create_user(username=f"testuser{i}", password="testpassword")
+            UserTag.objects.create(user=user, post=post)
```
Comment (ejariza-evowill) [inline blog/tests.py]: I would refactor these tests into a new class, so all the tags and post creation can be done only once at the `__init__` method.

**Review by ejariza-evowill — CHANGES_REQUESTED** [review]

File: blog/tests.py (hunk 2 above)
Comment (Magnico) [reply blog/tests.py]: to confirm, what you mean is to separate the tests test_post_tagged_count and test_post_last_tag_date in another class apart from PostTestCase, and use __init__ to create the user tags to be used in the two tests?

Comment (ejariza-evowill) [reply blog/tests.py]: yes, so you keep DRY and not have to repeat the process of setting up. But thinking about this I think is most appropriated to use the [setup method](https://docs.djangoproject.com/en/4.1/topics/testing/overview/#writing-tests) for this.

**Review by ejariza-evowill — APPROVED** [review]

**Comment by Magnico** [comment]
Changes made:
- Created test class UserTagTestCase

**Review by ejariza-evowill — APPROVED** [review]

**Pull request merged**